ddl/ddl_api.go

```diff
@@ -3396,9 +3396,10 @@ func (d *ddl) DropIndex(ctx sessionctx.Context, ti ast.Ident, indexName model.CI
 		return err
 	}
 
+	// Check for drop index on auto_increment column.
 	cols := t.Cols()
 	for _, idxCol := range indexInfo.Columns {
-		if mysql.HasAutoIncrementFlag(cols[idxCol.Offset].Flag) {
+		if mysql.HasAutoIncrementFlag(cols[idxCol.Offset].Flag) && countOfIndexContainColumn(t.Meta(), idxCol.Name.L) < 2 {
 			return autoid.ErrWrongAutoKey
 		}
 	}
```

Comment:
Could we extract this code(line3400-3405) directly into a function instead of using `countOfIndexContainColumn `?

Reply:
Great, Done.